### PR TITLE
Features/sql db migration

### PR DESCRIPTION
**Pull request by AlexSpodarev**
Hello @tommybrecher, 
please be kind to review my code of the latest migration to SQLITE3 platform.

Thanks

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,48 +2,37 @@
 All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
-## [0.0.4] - 2018-12-28
+## [0.1.1] - 2019-01-04
 
 ### Added
-- Database extended from 2 to 6 columns - [@AlexSpodarev](https://github.com/AlexSpodarev)
+- Database passed from .csv to SQLite3 - [@AlexSpodarev](https://github.com/AlexSpodarev)
+- URL now can be for example "api/seat" or "api/seat?model=leon&year=2015" - [@AlexSpodarev](https://github.com/AlexSpodarev)
```
Comment (tommybrecher) [inline CHANGELOG.md]: A better description would be:
- Added support for GET requests with query arguments, up to you if to fix or not

File: REST_API.py [hunk 2]
```diff
@@ -1,173 +1,171 @@
...
 @app.route('/api/<data>', methods=['POST'])
 def api_post(data):
     """
     When the request method is an HTTP/POST, the logic is to grab the data
     Check if it's in the DB, if it's not, we append it to the DB
-    TODO: Duplicates/Case/Error handling should be added
     """
-    if data.split('/')[0].isalpha():
+    input_params = request.get_json()
+    if len(input_params) is not (DB_SIZE - 1):
```
Comment (tommybrecher) [inline REST_API.py]: Like

File: REST_API.py [hunk 3]
```diff
@@ -1,173 +1,171 @@
...
-def fetch_from_db(key=None, db='database.csv', db_mode='r'):
+def append_to_db(key=None, request_dict=None):
     """
-    We pass in a key which starts out as None for error handling
-    the key is the Vendor of the car, we then open the DB
-    and check if the key matches the vendor column, if it does
-    we format the string and return it to the function which called
+    The variable content starts as None which is the recommended way according to google code style.
+    if content is None, a log message is printed with the content and type(content)
+    The parameters db and db_mode are initialized to database.csv and a (append) respectively,
```
Comment (tommybrecher) [inline REST_API.py]: No more database.csv ..

Also no more params named db and db_mode, please fix.

File: REST_API.py [hunk 4]
```diff
@@ -1,173 +1,171 @@
...
-            return jsonify(output)
     else:
-        """
-        If we didn't get a key, reject the request with a 404 not found
-        """
-        reject_invalid_request()
+        db = open_db_connection()
+        cursor = db.cursor()
 
+        param = f' \'{key}\' '
```
Comment (tommybrecher) [inline REST_API.py]: Not clear, please change to 
param = f"'{key}'"

File: REST_API.py [hunk 5]
```diff
@@ -1,173 +1,171 @@
...
+        for k, v in request_dict.items():
+            param += f",'{v}'"
```
Comment (tommybrecher) [inline REST_API.py]: Please add a comment for this whole block, the logic is not straight forward with += and is kinda confusing

File: REST_API.py [hunk 6]
```diff
@@ -1,173 +1,171 @@
...
+                        VALUES ({param});"
 
-            else:
-                app.logger.info(f"Couldn't find '{content}' in '{db}'")
-                return result
-        return result
+        """Adding content """
+        cursor.execute(query)
 
+        """Closing DB"""
```
Comment (tommybrecher) [inline REST_API.py]: You are not closing the DB,  please remove those comments and add a description of the logic used in this function right under the function declaraton instead

**Review by tommybrecher — CHANGES_REQUESTED** [review]
Hi Alex, there are a few changes to be made,

FYI

File: CHANGELOG.md (hunk 1 above)
Comment (AlexSpodarev) [reply CHANGELOG.md]: Totally agree, i've fixed the description.

**Comment by AlexSpodarev** [comment]
I've fixed the issues and remastered description part throughout the code, 
please review the version @tommybrecher .

**Review by tommybrecher — APPROVED** [review]
LGTM!

**Pull request merged**